assets/javascripts/render.js

```diff
@@ -23,12 +23,15 @@ function createElement(tag, content = [], attrs = {}) {
 }
 
 function renderTemplate(template, args = {}) {
-    for (let key in args) {
-        template = template.split('$' + key + '$').join(args[key]);
-        template = template.split(encodeURIComponent('$' + key + '$'));
-        template = template.join(encodeURIComponent(args[key]));
+    if (!template) {
+        return '';
+    }
```

Comment:
I don't understand. What is a "template snippet"? And what should be the cases where it is ok that this is not existant? Can you add a test for the new code branch?

Reply:
Reading the code will answer your first question. (I also just read the code as it hasn't been written by me.)

I wasn't sure about the cases when creating this PR but comments in https://progress.opensuse.org/issues/72289 show at least one case. I could try to cover that case or simply invoke the JavaScript function directly with some fake data via `execute_script` but I don't think it is worth it at this point. (We should rather invest the time to get JavaScript coverage analysis and then generally improve the tests.)

Reply:
https://progress.opensuse.org/issues/73084